### PR TITLE
Use BinaryData.FromBytes in ModelSerializationExtensions

### PR DESCRIPTION
**Pull request by eerhardt**
# Description

Using JsonSerializer to serialize a hard coded string is heavier weight than necessary. We can hard-code the UTF8 bytes directly.

This allows (parts of) the OpenAI library to be used in a native AOT'd app.

Fix https://github.com/microsoft/typespec/issues/4282

# Checklist

To ensure a quick review and merge, please ensure:
- [x] The PR has a understandable title and description explaining the _why_ and _what_.
- [x] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [x] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [x] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [x] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Common/Output/Models/Types/HelperTypeProviders/ModelSerializationExtensionsProvider.cs [hunk 1]
```diff
@@ -51,7 +51,9 @@ public ModelSerializationExtensionsProvider() : base(Configuration.HelperNamespa
                     type: typeof(BinaryData),
                     name: _sentinelBinaryDataName)
                 {
-                    InitializationValue = BinaryDataExpression.FromObjectAsJson(Literal("__EMPTY__"))
+                    InitializationValue = BinaryDataExpression.FromBytes(
+                        new InvokeInstanceMethodExpression(
+                            LiteralU8("\"__EMPTY__\""), "ToArray", [], null, false))
```
Comment (KrzysztofCwalina) [inline src/AutoRest.CSharp/Common/Output/Models/Types/HelperTypeProviders/ModelSerializationExtensionsProvider.cs:56]: would it make sense to add some comments here to describe what's happening here.

Comment (ArcturusZhang) [reply src/AutoRest.CSharp/Common/Output/Models/Types/HelperTypeProviders/ModelSerializationExtensionsProvider.cs:56]: So our generator expressions currently do not have the ability of adding comments.
We could add the comments in the generator.
If we want to add comments in generated code, we might have troubles.

**Review by ArcturusZhang — APPROVED** [review]

File: src/AutoRest.CSharp/Common/Output/Models/Types/HelperTypeProviders/ModelSerializationExtensionsProvider.cs (hunk 1 above)
Comment (eerhardt) [reply src/AutoRest.CSharp/Common/Output/Models/Types/HelperTypeProviders/ModelSerializationExtensionsProvider.cs:58]: I've added a quick comment explaining why we are using FromBytes and hard-coding the bytes directly.

**Review by KrzysztofCwalina — APPROVED** [review]

**Comment by KrzysztofCwalina** [comment]
@m-nash, do you have any other concerns or we can merge?

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**